content/en/docs/Architecture/cloud-init.md

```diff
@@ -41,32 +41,51 @@ stages:
+| _after-install_            | This stage runs after active,passive and recovery images are installed and after disks have been encrypted                                                                                                                                                                          |
+| _before-reset_             | This stage happens after partitions have been formatted and mounted but before the image has been reset                                                                                                                                                                          |
+| _after-reset_              | This stage happens after partitions have been formatted and mounted and active and passive images reset                                                                                                                                                                          |
+| _after-reset-chroot_       | This stage happens after active has been reset but before passive has been touched.                                                                                                                                                                                                 |
+| _before-upgrade_           | This stage happens after mounting partitions with RW but before any image has been upgraded                                                                                                                                                                                         |
+| _after-upgrade_            | This stage happens after upgrade has been done                                                                                                                                                                                                                                      |
+| _after-upgrade-chroot_     | This stage happens after the image has been upgraded                                                                                                                                                                                                                                |
 
 Note: Steps executed at the `chroot` stage are running inside the new OS as chroot, allowing to write persisting changes to the image, for example by downloading and installing additional software.
 
+{{% alert title="About after and before substages" color="info" %}}
+All the above stages are automatically run with a `before` and `after` suffix automatically, so users can always run something before of after the given
+stage with ease. So if a configuration is doing something in the `before-install` stage and you want to runs something before it, you could use the `before-install.before` to run it before the main stage runs.
```

Comment:
this is exactly why I think it would make sense to rename `before-install` doesn't really tell me at which point because it means a lot of things and `before-install.before_` and `before_install.after_` are just weird imo :P

Reply:
tbh I think this is wrong and I fucked up. I mean, if you run them manually via the agent they do but in here I think we just run the exact stage and nothing else, so I migth need to remove this.